### PR TITLE
Deprecate MultiTableMetadata in favor for Metadata

### LINKED ISSUE #2129: Enable multi table synthesizers to use new Metadata
### Problem Description

_As a user, I'd like to be able to use my metadata wherever it is applicable once created without having to do extra work._

### Expected behavior

Once #2104 is complete, we need to enable multi table synthesizers to work with the new Metadata
- Support the new class on initialization
- Deprecate MultiTableMetadata
    - Raise a future warning if users use this
- They should be backwards compatible (still support the old class)

### Additional context

Since the new class will be almost identical to the current MultiTableMetadata class, this should be a straightforward fix.

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #2129
CU-86b19awxk

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b19awxk SDV - Enable multi table synthesizers to use new Metadata #2129](https://app.clickup.com/t/86b19awxk)

File: sdv/metadata/multi_table.py [hunk 1]
```diff
@@ -914,6 +914,47 @@ def get_table_metadata(self, table_name):
         self._validate_table_exists(table_name)
         return deepcopy(self.tables[table_name])
 
+    def anonymize(self):
```
Comment (amontanez24) [inline sdv/metadata/multi_table.py:917]: why is it treating this as new code?

**Review by amontanez24 — COMMENTED** [review]
I think we need to either rebase this off the target branch or rebase the target branch off of main

**Comment by lajohn4747** [comment]
Will try to rebase entirely to avoid extra commits

**Pull request closed without merging**